lib/matplotlib/colors.py

```diff
@@ -960,6 +960,229 @@ def scaled(self):
         return (self.vmin is not None and self.vmax is not None)
 
 
+class FuncNorm(Normalize):
+    """
+    Creates a normalizer using a custom function
+
+    The normalizer will be a function mapping the data values into colormap
+    values in the [0,1] range.
+    """
+
+    def __init__(self, f, finv=None, **normalize_kw):
+        """
+        Specify the function to be used, and its inverse, as well as other
+        parameters to be passed to `Normalize`. The normalization will be
+        calculated as (f(x)-f(vmin))/(f(max)-f(vmin)).
+
+        Parameters
+        ----------
+        f : callable or string
+            Function to be used for the normalization receiving a single
+            parameter, compatible with scalar values and ndarrays.
+            Alternatively a string from the list ['linear', 'quadratic',
```

Comment:
Yes, this is precisely why I did not do it. I do not want to encourage people to have any direct contact _StringFuncParser. They should just see the result, so at least we can still switch to something else in the future if something better than that comes up.

The best way to solve this would probably be to expose the available strings through a public helper function acting as an interface between _StringFuncParser and the values. Something like `GetValidStringFuncs`. But for now I think it may just be better to do it by hand.